src/impl/tpls/KokkosBlas2_gemv_tpl_spec_decl.hpp

```diff
@@ -44,9 +44,456 @@
+extern "C" void sgemv_( const char* trans,
+                        const int* M, const int* N,
+                        const float* alpha,
+                        const float* A, const int* LDA,
+                        const float* X, const int* INCX,
+                        const float* beta,
+                        float* Y, const int* INCY);
+extern "C" void zgemv_( const char* trans,
+                        const int* M, const int* N,
+                        const std::complex<double>* alpha,
+                        const std::complex<double>* A, const int* LDA,
+                        const std::complex<double>* X, const int* INCX,
+                        const std::complex<double>* beta,
+                        std::complex<double>* Y, const int* INCY);
+extern "C" void cgemv_( const char* trans,
+                        const int* M, const int* N,
+                        const std::complex<float>* alpha,
+                        const std::complex<float>* A, const int* LDA,
+                        const std::complex<float>* X, const int* INCX,
+                        const std::complex<float>* beta,
+                        std::complex<float>* Y, const int* INCY);
+
 namespace KokkosBlas {
 namespace Impl {
+
+#define KOKKOSBLAS2_DGEMV_BLAS( LAYOUTA, LAYOUTX, LAYOUTY, MEM_SPACE, ETI_SPEC_AVAIL ) \
+template<class ExecSpace> \
+struct GEMV< \
+     Kokkos::View<const double**, LAYOUTA, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+                  Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
+     Kokkos::View<const double*, LAYOUTX, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+                  Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
+     Kokkos::View<double*, LAYOUTY, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+                  Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
+     true, ETI_SPEC_AVAIL> { \
+  typedef double SCALAR; \
+  typedef Kokkos::View<const SCALAR**, LAYOUTA, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > AViewType; \
+  typedef Kokkos::View<const SCALAR*, LAYOUTX, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > XViewType; \
+  typedef Kokkos::View<SCALAR*, LAYOUTY, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > YViewType; \
+ \
+  static void \
+  gemv (const char trans[], \
+        typename AViewType::const_value_type& alpha, \
+        const AViewType& A, \
+        const XViewType& X, \
+        typename YViewType::const_value_type& beta, \
+        const YViewType& Y) { \
+    \
+    Kokkos::Profiling::pushRegion("KokkosBlas::gemv[TPL_BLAS,double]"); \
+    const int M = static_cast<int> (A.extent(0)); \
+    const int N = static_cast<int> (A.extent(1)); \
+    constexpr int one = 1; \
+    int strides[2]; \
+    bool A_is_lr = std::is_same<Kokkos::LayoutRight,LAYOUTA>::value; \
+    A.stride(strides); \
+    const int LDA = strides[A_is_lr?0:1]; \
```

Comment:
LDA must never be zero, even if e.g., the number of rows is zero.  See e.g., http://github.com/trilinos/Trilinos/issues/3235 .